src/main/java/net/neoforged/neoforge/client/entity/animation/json/JsonAnimationParser.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.client.entity.animation.json;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import it.unimi.dsi.fastutil.Pair;
+import java.util.Map;
+import net.minecraft.client.animation.AnimationChannel;
+import net.minecraft.client.animation.AnimationDefinition;
+import net.minecraft.client.animation.Keyframe;
+import net.minecraft.client.animation.KeyframeAnimations;
+import net.minecraft.util.GsonHelper;
+import org.joml.Vector3f;
+
+/**
+ * A parser for parsing JSON entity animation files.
+ */
+public final class JsonAnimationParser {
+    @FunctionalInterface
+    private interface KeyframeVecFactory {
+        Vector3f apply(float x, float y, float z);
+    }
+
+    private static final Map<String, Pair<AnimationChannel.Target, KeyframeVecFactory>> TARGETS = Map.of(
+            "position", Pair.of(AnimationChannel.Targets.POSITION, KeyframeAnimations::posVec),
+            "rotation", Pair.of(AnimationChannel.Targets.ROTATION, KeyframeAnimations::degreeVec),
+            "scale", Pair.of(AnimationChannel.Targets.SCALE, KeyframeAnimations::scaleVec));
```

Comment:
I understand the purpose of this, but what if someone wanted to use the raw vector values, or some other transformer?